Code/ForceField/Wrap/ForceField.cpp

```diff
@@ -123,6 +123,61 @@ PyObject *ForceFieldGetExtraPointLoc(PyForceField *self, unsigned int idx) {
   return res;
 }
 
+double PyForceField::calcEnergyWithPos(const python::object &pos) {
+  PRECONDITION(this->field, "no force field");
+  if (pos != python::object()) {
+    int s = this->field->dimension() * this->field->numPoints();
+    int numElements = python::extract<int>(pos.attr("__len__")());
+    PRECONDITION(s == numElements,
```

Comment:
This could raise a `ValueError` instead of using the `PRECONDITION`
(occurs a few more times below)